lib/src/analysis.dart

```diff
@@ -339,7 +334,7 @@ class HumanErrorFormatter extends ErrorFormatter {
   void formatError(
       Map<AnalysisError, LineInfo> errorToLine, AnalysisError error) {
     var source = error.source;
-    var location = errorToLine[error].getLocation(error.offset);
+    var location = errorToLine[error]!.getLocation(error.offset);
```

Comment:
I only see one place where `formatError` is being used (on line 278), and in that location it would be much better to just pass the `LineInfo` in to this method directly than to pass in a map. In fact, in that location it looks like you could stop computing the map at all and just format the errors as you iterate over them the first time.